extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ArcEndpointRecorder.java

```diff
@@ -0,0 +1,131 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Default;
+
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.arc.InjectableObserverMethod;
+import io.quarkus.arc.impl.ArcContainerImpl;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.vertx.http.runtime.devmode.Json.JsonArrayBuilder;
+import io.quarkus.vertx.http.runtime.devmode.Json.JsonObjectBuilder;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+@Recorder
+public class ArcEndpointRecorder {
+
+    public Handler<RoutingContext> createBeansHandler() {
+        return new Handler<RoutingContext>() {
+
+            @Override
+            public void handle(RoutingContext ctx) {
+                ctx.response().putHeader("Content-type", "application/json");
+
+                ArcContainerImpl container = ArcContainerImpl.instance();
+                List<InjectableBean<?>> beans = container.getBeans();
+                beans.addAll(container.getInterceptors());
+
+                String kindParam = ctx.request().getParam("kind");
+                InjectableBean.Kind kind = kindParam != null ? InjectableBean.Kind.valueOf(kindParam) : null;
+                String scopeEndsWith = ctx.request().getParam("scope");
+                String beanClassStartsWith = ctx.request().getParam("beanClass");
```

Comment:
What is the rationale of having `scope` being used with `endsWith`  and `beanClass` used `startsWith`?

Reply:
I believe that it's practical. For scopes, users will usually pass built-in scope names. So it's more convenient to specify `RequestScoped` rather than `javax.enterprise.context.RequestScoped`. Also simple names of scopes do not usually collide. For bean classes however, it's very often useful to filter beans from a specific package, i.e. `beanClass=org.acme`.

Reply:
That's fine with me, just curious. It should however be documented

Reply:
Sure. Just finished the docs and a test ;-).